internal/ansible/ansible.go

```diff
@@ -175,9 +175,11 @@ func (p Parameters) playbookCmdFunc(ctx context.Context, playbookName string, pa
 // roleCmdFunc mimics https://github.com/operator-framework/operator-sdk/blob/707240f006ecfc0bc86e5c21f6874d302992d598/internal/ansible/runner/runner.go#L92-L118
 func (p Parameters) roleCmdFunc(ctx context.Context, roleName string, path string) cmdFuncType {
 	return func(behaviorVars map[string]string, checkMode bool) *exec.Cmd {
-		cmdArgs := []string{"run", filepath.Join(path, roleName)}
+		cmdArgs := []string{"run", p.WorkingDirPath}
 		cmdOptions := []string{
 			"--role", roleName,
+			"--roles-path", path,
+			"--project-dir", p.WorkingDirPath,
```

Comment:
do we need `--project-dir` here ?

PTAL:
```
  --project-dir PROJECT_DIR
                        optional path for the location of the playbook content directory (default=<private_data_dir/project)
```

Reply:
I got this error in the runtime `ERROR! the playbook: ansibleDir/4eb34f4b-4f77-4676-8371-5a51da41de22/091f98dd17e347c683efd0217adfac0b could not be found` 

looks like `--project-dir` flag is disturbing  the execution.

Reply:
I was having issues with the role finding the credentials before I added this, let me try backing it out and see if there is another setting I need to set to get that working

Reply:
I removed --project-dir and confirmed that I can reference credentials from the parent directory instead it seems to work, like this `ansible_ssh_private_key_file: ../ssh_id`

Reply:
@eljohnson92 I played with `--project-dir ` flag and it is effective for both roles and playbooks. So I believe that we should set it so your credentials will be referenced from ansibleRun specific dir  (`ansibleDir/UUID`)  and not the parent dir (`ansibleDir/`)  and not the parent. 

If I map ansible-runner default context to our provider context, we should get this mapping table:
`private_data_dir`  ->  `p.WorkingDirPath`
`private_data_dir/project`  (project-dir)  ->  `p.WorkingDirPath`

Reply:
@fahedouch just to ensure I understand, should I revert my latest change so we are again setting the `--project-dir p.WorkingDirPath `? do you have a workaround for your above error with the playbook?

Reply:
Yes please. Could you please revert your last changes to set again `--project-dir`. 
The error was related to my environnement. Please do not consider lt.
I am also thinking about making `p.WorkingDirPath` structure same as default ansible structure so we avoid setting custom path for context metadata (e.g project, etc..)
@eljohnson92 @morningspace

Reply:
that would mean you would be mounting stuff to the defaults mentioned in these options right?
```
  --artifact-dir ARTIFACT_DIR
                        optional path for the artifact root directory (default=<private_data_dir>/artifacts)
  --project-dir PROJECT_DIR
                        optional path for the location of the playbook content directory (default=<private_data_dir/project)
  --inventory INVENTORY
                        optional path for the location of the inventory content directory (default=<private_data_dir>/inventory)
```

Reply:
@eljohnson92 Yes.

Reply:
I think that is a good strategy, I updated this PR to work as things stand today, but we could always remove the --project-dir once that is fixed in the future

Reply:
@eljohnson92 I created a dedicated ticket https://github.com/crossplane-contrib/provider-ansible/issues/166